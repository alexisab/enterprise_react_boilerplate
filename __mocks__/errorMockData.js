const adminList = {
  admins: [
    {
      id: 1,
      name: "admin1",
      email: "admin1@eg.com",
      designation: null,
      status: "active",
      admin: true,
      super_admin: false,
      errors: [{ key: "error", message: "error message" }]
    },
    {
      id: 2,
      name: "admin2",
      email: "admin2@eg.com",
      designation: null,
      status: "active",
      admin: true,
      super_admin: false,
      errors: []
    }
  ]
};

const admin = adminList.admins[0];

const sendInviteMutate = {
  sendInviteMutate: jest
    .fn()
    .mockReturnValueOnce(Promise.resolve({ data: { sendInvite: admin } }))
};

const errorMockData = {
  admins: adminList,
  sendInvite: sendInviteMutate
};

export default errorMockData;
