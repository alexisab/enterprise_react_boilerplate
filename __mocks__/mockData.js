const adminList = {
  admins: [
    {id: 1, name: "admin1", email: "admin1@eg.com", designation: null, status: "active", admin: true, super_admin: false },
    {id: 2, name: "admin2", email: "admin2@eg.com", designation: null, status: "active", admin: true, super_admin: false }
  ]
};

const mockData = {
  admins: adminList,
};

export default mockData;
