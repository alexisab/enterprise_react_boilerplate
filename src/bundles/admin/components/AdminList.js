import React from 'react';
import { Table, Button } from 'antd';
import enhance from '../composers/AdminListComposer';

const AdminList = (props) => {
  const { data: { admins }, handleDelete, sendInvite } = props;
  console.log('admins', props);
  const columns = [{
    title: 'Name',
    dataIndex: 'name',
    key: 'name',
  }, {
    title: 'Email',
    dataIndex: 'email',
    key: 'email',
  }, {
    title: 'Status',
    dataIndex: 'status',
    key: 'status',
  }, {
    title: 'Action',
    key: 'action',
    render: (text, record) => (
      <div>
        <span>
          <Button type="danger" id={`delete-admin-${record.id}`} onClick={e => handleDelete(record.id)}>Delete User</Button>
        </span>
        { record.status === 'invited' &&
          <span>
            <Button type="primary ml-10" id={`invite-admin-${record.id}`} onClick={e => sendInvite(record.id)}>Send Invitation Again</Button>
          </span>
        }
      </div>
    ),
  }];

  return (
    <Table rowKey={record => `user-${record.id}`} dataSource={admins} columns={columns} />
  );
}

export default enhance(AdminList);
