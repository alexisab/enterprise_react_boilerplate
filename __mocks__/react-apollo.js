import React from "react";
import mockData from "./mockData";
import errorMockData from "./errorMockData";

const actualReactApollo = require.requireActual("react-apollo");

let mockProps = {};
let errorProps = false;

const setMockGraphQLProps = props => {
  mockProps = props;
};

const setMockGraphQLErrorProps = () => {
  errorProps = true;
};

const setMockedData = (mockDataSource, operation, queryValue) => {
  const mockedData = mockDataSource[queryValue];
  if (operation === "mutation") {
    setMockGraphQLProps(mockedData);
  } else {
    setMockGraphQLProps({ data: mockedData });
  }
};

const graphql = query => Component => props => {
  const { operation } = query.definitions[0];
  const queryValue = query.definitions[0].name.value;
  if (errorProps) {
    setMockedData(errorMockData, operation, queryValue);
  } else {
    setMockedData(mockData, operation, queryValue);
  }

  return <Component {...mockProps} {...props} />;
};

const {
  compose,
  createBatchingNetworkInterface,
  ApolloClient
} = actualReactApollo;

export {
  graphql,
  compose,
  setMockGraphQLProps,
  setMockGraphQLErrorProps,
  createBatchingNetworkInterface,
  ApolloClient
};
