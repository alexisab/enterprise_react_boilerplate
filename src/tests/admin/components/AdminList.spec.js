import React from "react";
import renderer from "react-test-renderer";
// import { shallow } from "enzyme";
import { MemoryRouter } from "react-router";
import AdminList from "../../../bundles/admin/components/AdminList";
import { adminList } from "../../../../__mocks__/mockData";

const handleDelete = jest.fn();
const sendInvite = jest.fn();

const props = {
  data: adminList,
  handleDelete,
  sendInvite,
  formData: {
    name: "",
    email: ""
  },
  formErrors: {
    name: "",
    email: ""
  }
};

describe("AdminList", () => {
  test("it should match snapshot", () => {
    const component = renderer.create(
      <MemoryRouter>
        <AdminList {...props} />
      </MemoryRouter>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
