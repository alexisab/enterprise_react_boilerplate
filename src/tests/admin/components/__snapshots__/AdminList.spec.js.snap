// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminList it should match snapshot 1`] = `
<div
  className="ant-table-wrapper"
  style={undefined}
>
  <div
    className="ant-spin-nested-loading"
    style={null}
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table ant-table-large ant-table-scroll-position-left"
        id={undefined}
        style={Object {}}
      >
        <div
          className="ant-table-content"
        >
          <div
            className="ant-table-body"
            onScroll={[Function]}
            style={Object {}}
          >
            <table
              className=""
              style={Object {}}
            >
              <colgroup>
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
              </colgroup>
              <thead
                className="ant-table-thead"
              >
                <tr
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <th
                    className=""
                  >
                    <span>
                      Name
                    </span>
                  </th>
                  <th
                    className=""
                  >
                    <span>
                      Email
                    </span>
                  </th>
                  <th
                    className=""
                  >
                    <span>
                      Status
                    </span>
                  </th>
                  <th
                    className=""
                  >
                    <span>
                      Action
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  className="ant-table-row  ant-table-row-level-0"
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDoubleClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    <span
                      className="ant-table-row-indent indent-level-0"
                      style={
                        Object {
                          "paddingLeft": "0px",
                        }
                      }
                    />
                    admin1
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    admin1@eg.com
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    active
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    <div>
                      <span>
                        <button
                          className="ant-btn ant-btn-danger"
                          id="delete-admin-1"
                          onClick={[Function]}
                          type="button"
                        >
                          <span>
                            Delete User
                          </span>
                        </button>
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  className="ant-table-row  ant-table-row-level-0"
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDoubleClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    <span
                      className="ant-table-row-indent indent-level-0"
                      style={
                        Object {
                          "paddingLeft": "0px",
                        }
                      }
                    />
                    admin2
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    admin2@eg.com
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    active
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    <div>
                      <span>
                        <button
                          className="ant-btn ant-btn-danger"
                          id="delete-admin-2"
                          onClick={[Function]}
                          type="button"
                        >
                          <span>
                            Delete User
                          </span>
                        </button>
                      </span>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <ul
        className="ant-pagination ant-table-pagination"
        style={Object {}}
        unselectable="unselectable"
      >
        <li
          aria-disabled={true}
          className="ant-pagination-disabled ant-pagination-prev"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title="Previous Page"
        >
          <a
            className="ant-pagination-item-link"
          />
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={1}
        >
          <a>
            1
          </a>
        </li>
        <li
          aria-disabled={true}
          className="ant-pagination-disabled ant-pagination-next"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title="Next Page"
        >
          <a
            className="ant-pagination-item-link"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;
